src/patient-registration/patient-registration.component.tsx

```diff
@@ -117,18 +169,24 @@ export const PatientRegistration: React.FC = () => {
         ],
       },
     };
-
-    savePatient(abortController, patient).then(
-      response => response.status == 201 && history.push(`/patient/${response.data.uuid}/chart`),
-      createErrorHandler(),
-    );
+    savePatient(abortController, patient)
+      .then(response => response.status == 201 && history.push(`/patient/${response.data.uuid}/chart`))
+      .catch(response => {
+        if (response.responseBody.error.globalErrors) {
+          response.responseBody.error.globalErrors.forEach(error => {
+            showToast({ description: error.message });
+          });
+        } else {
+          createErrorHandler();
+        }
+      });
```

Comment:
Nice! This seems like a big improvement. What sort of errors does this catch?

Reply:
@FlorianRappl explained to me back then that it depends on the browser's error API to handle any error that happened in the async call. See [thread](https://openmrs.slack.com/archives/CHP5QAE5R/p1594892146302400) in slack.

Reply:
Yes, sorry, I was just curious whether there were examples of actual errors that you were seeing getting caught. No worries.

Reply:
While the catch etc. is good the `createErrorHandler` should not be called *at all* somewhere. Maybe the implementation changed (then please forgive my comment), but in the implementation I know it should be treated like a singleton. `createErrorHandler` essentially couples to the browsers error handlers (e.g., onError) and thus would show multiple toasts if called (i.e., registered) multiple times.

It would make sense if the error handler would just register itself once and then would prevent further registrations.

As a result -> please remove the `createErrorHandler()`.

Reply:
Removed :)

Reply:
Nice catch, @FlorianRappl , it was about time I actually read the implementation of esm-error-handling :)

But just for future reference, `createErrorHandler` doesn't add the listener to `window.onError`. The listeners get added at the top level of `esm-error-handler`, so they can only happen once. What `createErrorHandler` does is (apparently, just reading [the code](https://github.com/openmrs/openmrs-esm-error-handling/blob/50053ebc76e0f9695f5731bc62b831be842d94c3/src/openmrs-esm-error-handling.ts#L29)) to ensure that the error contains a stack trace for the call site. So I guess its intended usage is like `.catch(createErrorHandler())`. Indeed, the way it had been used here wouldn't do anything.

Reply:
@brandones It would actually, because it was in a `.catch(...)` block.

Reply:
I did it that way with something in my mind like: 

If there was an error in the request and the server responded with the custom error message, toast it like:

```java

showToast({ description: error.message });

```

If there was no custom error message, well lets call the default: `createErrorHandler()`. 

All the above wrapped in the `.catch()`

Reply:
@samuelmale The following two blocks are different:

```javascript
.catch(createErrorHandler())
```

vs

```javascript
.catch(() => { createErrorHandler() })
```

the latter is more like what you've written. `createErrorHandler` is a function that returns a function that handles errors. You aren't doing anything with its return value.

Reply:
Let's definitely add that `else` block back—we don't want to swallow any errors. But maybe we can write it like

```javascript
} else {
  createErrorHandler()(response)
}
```

Reply:
This is the only blocker before I hit the approve button. The other things would be nice but I'm not going to block on them.

Reply:
Resolved